### ISSUE TITLE
[v4b3] Datagrid row cursor CSS

### ISSUE THREAD

**Issue by smeng9** [marmelab/react-admin#7407]
**What you were expecting:**

I hope for datagrid rows with `rowClick="expand"` but `isRowExpandable` evaluated to falsy does not show a `cursor:pointer` on that row, it should show the default cursor

**What happened instead:**

Currently even if the row is not expandable, there is still a pointer, which is confusing for some of user.
Also there seems is not a good way to override this besides pulling datagrid apart by implementing our own datagrid row logic just for a style change.

**Steps to reproduce:**

**Related code:**

* Preferably, a CodeSandbox forked from https://codesandbox.io/s/github/marmelab/react-admin/tree/master/examples/simple
* A link to a GitHub repo with the minimal codebase to reproduce the issue

```
<Datagrid
        rowClick="expand"
        expand={<Panel />}
        isRowExpandable={(row) => shouldExpand(row)}   // some of rows is not expandable but still shows a cursor: pointer
>
    {children}
</Datagrid>
```

**Other information:**

Although everything conforms to the `sx` documentation for `& .RaDatagrid-clickableRow`, the user experience may not be very good.

I hope we can change how the CSS API `sx` here https://github.com/marmelab/react-admin/blob/next/docs/Datagrid.md?plain=1#L692 to if rowClick is a truthy string or the function evaluated to be truthy

```
<Datagrid
        rowClick={(id, resource, record) => shouldExpand(record) ? "expand" : null}
        expand={<Panel />}
--     /* isRowExpandable={(row) => shouldExpand(row)} */   // we probably even can deprecate the need for this prop
>
    {children}
</Datagrid>
```

This would require us to change this line https://github.com/marmelab/react-admin/blob/next/packages/ra-ui-materialui/src/list/datagrid/DatagridBody.tsx#L43 `[DatagridClasses.clickableRow]: rowClick`

So we check if `rowClick` is a function and we call the function to see the return value to determine whether this row is clickable.

In this way we will not show a cursor: pointer for non-clickable row.

**Environment**

* React-admin version: v4b3
* Last version that did not exhibit the issue (if applicable):
* React version: 18.0.0-rc.2
* Browser: Edge 99
* Stack trace (in case of a JS error):

**Comment by fzaninotto** [comment]
Fixed by #7411

**Comment by WiXSL** [comment]
> Also there seems is not a good way to override this besides pulling datagrid apart by implementing our own datagrid row logic just for a style change.

You can do something like this:

```jsx
 <Datagrid
    sx={{
        '& .RaDatagrid-clickableRow': { cursor: 'default' },
    }}
>
...
</Datagrid>
```

**Comment by smeng9** [comment]
Hi @WiXSL, Thanks for your suggestion.

However, the sx here applied to all the rows, not per row basis on some functions that accepts a record.

The use case here needs to make some row clickable and others don't based on returned data. The `pointer:cursor` need to match with that case.

**Comment by fzaninotto** [comment]
Fixed by #7468
